fs/vfs/main.cc

```diff
@@ -2345,6 +2345,26 @@ ssize_t sendfile(int out_fd, int in_fd, off_t *_offset, size_t count)
 #undef sendfile64
 LFS64(sendfile);
 
+extern "C" OSV_LIBC_API
+ssize_t copy_file_range(int fd_in, off_t *off_in,
+                        int fd_out, off_t *off_out,
+                        size_t len, unsigned int flags)
+{
+    //Non-zero flags are rejected according to the manual
+    if (flags != 0) {
+        errno = EINVAL;
+        return -1;
+    }
+    //We do not support writing to a file at specified offset because
+    //we delegate to sendfile() which assumes current position of the output
+    //file
+    if (off_out) {
+        errno = EINVAL;
```

Comment:
Maybe add UNIMPLEMENTED(...) message?

Reply:
I am not sure we want to `abort()` in this case, as I think in this scenario an app may simply fall back to other libc functions. So returning `-1` may be better in this case. But maybe we can add WARN here.

Reply:
Right, I mis-remembered that UNIMPLEMENTED only warned.
Perhaps the most appropriate command is WARN_ONCE() - so if some application uses this function in a loop, we'll only see the warning once?